src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -37,6 +42,81 @@ const ChannelListScreenV2 = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
+
+  React.useEffect(() => {
+    const checkNotif = async () => {
+      if (!navigation.isFocused()) return;
+      PushNotification.checkPermissions(
+        (
+          data: PushNotificationPermissions & {
+            authorizationStatus?: number;
+          }
+        ) => {
+          const lastPromptTime = StorageUtils.lastPromptNotification.get();
+
+          // For testing purpose,  (20000 milliseconds)
+          const promptInterval = 20000;
+          const currentTime = new Date().getTime();
+          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+            return; // Don't show the prompt if the interval has not passed
+          }
+
+          const needsPermission =
+            data.alert === false || data.badge === false || data.sound === false;
+
+          const showAlert = (onPressAction: () => void) =>
+            Alert.alert(
+              "Don't Miss New Message",
+              'Allow notification to know when friends send you messages',
+              [
+                {text: 'Not now', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
+                {text: 'Allow', onPress: onPressAction}
+              ]
+            );
+
+          switch (data.authorizationStatus) {
+            case 1:
+              if (needsPermission) {
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              } else {
+                requestPermission();
+              }
+              break;
+            case 2:
+              if (needsPermission) {
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              }
+              break;
+            case 3:
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => requestPermission());
+              break;
+            default:
+              break;
+          }
+        }
+      );
+    };
+
+    checkNotif();
+  }, [navigation?.isFocused?.()]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `useEffect` hook, you are checking permissions and showing an alert based on the authorization status. However, there's a lot of repeated code here that could be simplified. For example, the `showAlert` function and the setting of `lastPromptNotification` are repeated multiple times. You could refactor this to reduce redundancy and improve readability.
```diff
  React.useEffect(() => {
    const checkNotif = async () => {
      if (!navigation.isFocused()) return;
      PushNotification.checkPermissions(
        (
          data: PushNotificationPermissions & {
            authorizationStatus?: number;
          }
        ) => {
          const lastPromptTime = StorageUtils.lastPromptNotification.get();

          // For testing purpose,  (20000 milliseconds)
          const promptInterval = 20000;
          const currentTime = new Date().getTime();
          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
            return; // Don't show the prompt if the interval has not passed
          }

          const needsPermission =
            data.alert === false || data.badge === false || data.sound === false;

          const showAlert = (onPressAction: () => void) =>
            Alert.alert(
              "Don't Miss New Message",
              'Allow notification to know when friends send you messages',
              [
                {text: 'Not now', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
                {text: 'Allow', onPress: onPressAction}
              ]
            );

          if (needsPermission) {
            StorageUtils.lastPromptNotification.set(currentTime.toString());
            showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
          } else if (data.authorizationStatus === 3) {
            StorageUtils.lastPromptNotification.set(currentTime.toString());
            showAlert(() => requestPermission());
          } else {
            requestPermission();
          }
        }
      );
    };

    checkNotif();
  }, [navigation?.isFocused?.()]);
```
This refactored code reduces redundancy by only calling `showAlert` and setting `lastPromptNotification` once when `needsPermission` is true. It also simplifies the switch statement by directly checking the `authorizationStatus` value.


<!-- This is an auto-generated comment by OSS CodeRabbit -->